components/compliance-service/migrations/migration_workflow.go

```diff
@@ -0,0 +1,132 @@
+		ESClient: client,
+		Upgrades: upgrades,
+	}, cereal.TaskExecutorOpts{Workers: workerCount})
+
+}
+
+type MigrationWorkflow struct {
+}
+
+type MigrationWorkflowParameters struct {
+	DayLatestFlag    bool
+	ControlIndexFlag bool
+}
+
+type MigrationWorkflowPayload struct {
+	DayLatestFlag    bool
+	ControlIndexFlag bool
+}
+
+func (s *MigrationWorkflow) OnStart(w cereal.WorkflowInstance,
+	ev cereal.StartEvent) cereal.Decision {
+
+	logrus.Debug("In control-workflow start method")
+
+	workflowParams := MigrationWorkflowParameters{}
+	err := w.GetParameters(&workflowParams)
+	if err != nil {
+		err = errors.Wrap(err, "failed to unmarshal control-workflow parameters")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+
+	logrus.Debugf("In On Start Method %s", workflowParams.DayLatestFlag)
+
+	workflowPayload := MigrationWorkflowPayload{
+		DayLatestFlag: workflowParams.DayLatestFlag,
+	}
+
+	err = w.EnqueueTask(DayLatestMigrationTaskName, GenerateDayLatestMigrationParameters{
+		DayLatestFlag: workflowParams.DayLatestFlag,
+	})
+	if err != nil {
+		err = errors.Wrap(err, "failed to enqueue the control-task")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+	return w.Continue(&workflowPayload)
+}
+
+func (s *MigrationWorkflow) OnTaskComplete(w cereal.WorkflowInstance,
+	ev cereal.TaskCompleteEvent) cereal.Decision {
+
+	var payload MigrationWorkflowPayload
+	if err := w.GetPayload(&payload); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal control-workflow payload")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+
+	logrus.Debugf("Entered ControlWorkflow > OnTaskComplete with payload %+v", payload)
+
+	//get the results returned by task run
+	var dayLatestParameters GenerateDayLatestMigrationParameters
+	err := ev.Result.Get(&dayLatestParameters)
+	if err != nil {
+		err = errors.Wrap(err, "failed to get the task run result in OnTaskComplete")
+		logrus.WithError(err).Error()
+		return w.Fail(err)
+	}
+
+	return w.Complete()
+}
+
+func (s *MigrationWorkflow) OnCancel(w cereal.WorkflowInstance, ev cereal.CancelEvent) cereal.Decision {
+	logrus.Debugf("ReportWorkflow got OnCancel")
+	return w.Complete()
+}
+
+type GenerateDayLatestMigrationTask struct {
+	ESClient *ingestic.ESClient
+	Upgrades *Upgrades
+}
+
+type GenerateDayLatestMigrationParameters struct {
+	DayLatestFlag bool
+}
+
+func (t *GenerateDayLatestMigrationTask) Run(ctx context.Context, task cereal.Task) (interface{}, error) {
+
+	var job GenerateDayLatestMigrationParameters
+	if err := task.GetParameters(&job); err != nil {
+		err = errors.Wrap(err, "could not unmarshal GenerateReportParameters")
+		logrus.WithError(err).Error()
+		return nil, err
+	}
+
+	logrus.Info("Inside the daily Latest Flag upgrades")
+	err := t.ESClient.SetNodesDayLatestFalse(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not unmarshal GenerateReportParameters")
+	}
+	t.Upgrades.UpdateDayLatestFlagToTrue()
```

Comment:
there is a possibility to get error for this stmt, we are not handling here.

Reply:
done